### PR TITLE
support copy result in STS

### PR DESCRIPTION
**Pull request by alanrenmsft**
Handle the copy results to clipboard in STS for optimized performance.  I tested all the selection patterns and also with large result set.

Test data: 50k rows with 40 columns
Before (ADS loads all data to UI side and then copy) - ~2 minutes with UI response delay.
Now: less than 10 seconds.

### CODE REVIEW

**Comment by alanrenmsft** [comment]
@kburtram there is no dotnet built-in support to interact with clipboard, and this is the most popular package out there, any concerns?

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/CopyResultsRequest.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        /// <summary>
+        /// Whether to remove the line break from cell values.
+        /// </summary>
+        public bool RemoveNewLines { get; set; }
+
+        /// <summary>
+        /// Whether to include the column headers.
+        /// </summary>
+        public bool IncludeHeaders { get; set; }
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/CopyResultsRequest.cs:33]: is this going to support "Copy only headers" like discussed in this issue https://github.com/microsoft/azuredatastudio/issues/8902?

**Review by kburtram — APPROVED** [review]
The TextCopy package looks popular/broadly used & is actively maintained.  Since it's MIT it is a pre-approved license for use in this project.  We should update the 3rd party notices file in ADS & vscode-mssql to include the license for this package though.

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/CopyResultsRequest.cs (hunk 1 above)
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/CopyResultsRequest.cs:33]: this property is to control whether the column names should be included along with the selected cells. the copy header action is a separate thing and we can already get it from the result metadata directly in ADS.

image

**Pull request merged**